File: Python/Implement Stack using Queues.py
class Stack:
    # initialize your data structure here.
    def __init__(self):
        self.list_1 = []
        self.list_2 = []

    # @param x, an integer
    # @return nothing
    def push(self, x):
        self.list_1.append(x)
        while len(self.list_2):
            self.list_1.append(self.list_2.pop(0))
        self.list_2 = self.list_1
        self.list_1 = []

    # @return nothing
    def pop(self):
        if len(self.list_2):
            self.list_2.pop(0)
        
    # @return an integer
    def top(self):
        if len(self.list_2):
            return self.list_2[0]

    # @return an boolean
    def empty(self):
        return len(self.list_2) == 0

File: java/Implement_Stack_using_Queues.java
// Using one queue to implement stack.
// The main operation is move the front n-1 value to the back, which makes the last value on the head of the queue.
// Time is O(n) on pop and top, but O(1) on push, space is O(n).

class MyStack {
    // Push element x onto stack.
    Queue<Integer> que = new LinkedList<Integer>();
    public void push(int x) {
        que.offer(x);
    }

    // Removes the element on top of the stack.
    public void pop() {
        int size = que.size();
        for (int i = 0; i < size-1; ++i)
            que.offer(que.poll());
        que.poll();
    }

    // Get the top element.
    public int top() {
        int size = que.size();
        for (int i = 0; i < size-1; ++i)
            que.offer(que.poll());
        int val = que.peek();
        que.offer(que.poll());
        return val;
    }

    // Return whether the stack is empty.
    public boolean empty() {
        return que.peek() == null;
    }
}
